main.rb

```diff
@@ -0,0 +1,48 @@
+require_relative 'request'
+require_relative 'gist_content'
+
+loop do
+  puts "Ingrese la ruta"
+  pathname = gets.chomp
+  puts "Descripcion:"
+  description = gets.chomp
+
+  loop do
+    puts "Quieres el gist publico? Si/No"
+    state = gets.chomp.capitalize
+
+    if state == 'Si'
+      @state = true
+      break
+    elsif state == 'No'
+      @state = false
+      break
+    else
+      puts "Respuesta incorrecta"
+    end
+  end
+
+  values = Received_values.new(pathname, description, @state)
+  values.path_existence
+
+  if values.data
+    values.body
+    request = Assemble.new(description, @state, values.data)
+
+    begin
+      request.request
+    rescue SocketError => exception
+      p "Error de conexión"
+      p "Intentar de nuevo?"
```

Comment:
The same comment with the 'put', make your code in English or Spanish but not both